lib/rules/dot-notation.js

```diff
@@ -59,6 +62,23 @@ module.exports = {
                             message: "[{{propertyValue}}] is better written in dot notation.",
                             data: {
                                 propertyValue: JSON.stringify(node.property.value)
+                            },
+                            fix(fixer) {
+                                const leftBracket = sourceCode.getTokenBefore(node.property);
+                                const rightBracket = sourceCode.getTokenAfter(node.property);
+                                const textBeforeProperty = sourceCode.text.slice(leftBracket.range[1], node.property.range[0]);
+                                const textAfterProperty = sourceCode.text.slice(node.property.range[1], rightBracket.range[0]);
+
+                                if (textBeforeProperty.trim() || textAfterProperty.trim()) {
+
+                                    // Don't perform any fixes if there are comments inside the brackets.
+                                    return null;
+                                }
+
+                                return fixer.replaceTextRange(
+                                    [leftBracket.range[0], rightBracket.range[1]],
+                                    `${textBeforeProperty}.${node.property.value}${textAfterProperty}`
```

Comment:
Good point; this was originally there to ensure that comments didn't get deleted, but the rule now doesn't do the fix if there are comments anyway.